### DISCUSSION TITLE
/oo/code-init.html

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#50] [Announcements]
# /oo/code-init.html

Java程序员进阶之路，小白的零基础Java教程，认真聊聊 Java中的代码初始化块：对成员变量进行更复杂的赋值

https://tobebetterjavaer.com/oo/code-init.html

**Comment by DylanLJC** [comment]
感觉这个挺重要的：对象在初始化的时候会先调用构造方法，这是毫无疑问的，只不过，构造方法在执行的时候会把代码初始化块放在构造方法中其他代码之前

**Reply by DylanLJC** [reply]
类实例化的时候执行代码初始化块；
实际上，代码初始化块是放在构造方法中执行的，只不过比较靠前；
代码初始化块里的执行顺序是从前到后的。
